lib/usecasing/base.rb

```diff
@@ -7,22 +7,31 @@ def self.included(base)
     end
 
     module ClassMethods
+      def required_params(*params)
+        @parameters ||= []
+        @parameters.push(*params)
+        parameters
+      end
+
+      def parameters
+        concat_superclass_variable('parameters')
+      end
 
       def depends(*deps)
         @dependencies ||= []
         @dependencies.push(*deps)
       end
 
       def dependencies
-        return [] unless superclass.ancestors.include? UseCase::Base
-        value = (@dependencies && @dependencies.dup || []).concat(superclass.dependencies)
-        value
+        concat_superclass_variable('dependencies')
       end
 
       def perform(ctx = { })
         tx(ExecutionOrder.run(self), ctx) do |usecase, context|
           instance = usecase.new(context)
           instance.tap do | me |
+            me.required_params
```

Comment:
This should denote an action, probably a better name would be `extract_required_params`.

Reply:
Also, i would avoid using `require` (and any variation) since you don't enforce it, we are just copying the variables from the context to the usecase instance.

Reply:
Now the method raises an exception when a given parameter does not exist in the context so i think ```required_params``` is a good name now.